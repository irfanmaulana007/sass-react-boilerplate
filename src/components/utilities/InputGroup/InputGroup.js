import React from 'react'

import { TextSmall } from 'Components/utilities/Text/Text'

import './InputGroup.scss'

export default function InputGroup (props) {
    let inputType;

    switch (props.type) {
        case 'text':
            inputType = <input className={`px-2 py-1 font-light bg-field w-${props.width ? props.width + ' inline' : 'full block'} ${props.classadditional}`} {...props} onChange={props.onChange} />
            break
        case 'password':
            inputType = <input className={`px-2 py-1 font-light bg-field w-${props.width ? props.width + ' inline' : 'full block'} ${props.classadditional}`} {...props} onChange={props.onChange} />
            break
        case 'select':
            inputType = (
                <select className={`capitalize px-2 py-2 font-light bg-field ${props.width ? 'w-' + props.width + ' inline' : 'full block'} ${props.disabled ? 'bg-gray-200' : ''} ${props.classadditional}`} {...props} defaultValue="" onChange={props.onChange}>
                    { props.excludeselectoption ? '' :  <option value="">Select</option> }
                    {props.options.map((values, key) =>
                        <option key={key} value={values.id}>{values.name}</option>
                    )}
                </select>
            )
            break
        default:
            inputType = <input className={`px-2 py-1 font-light bg-field w-${props.width ? props.width + ' inline' : 'full block'} ${props.classadditional}`} {...props} onChange={props.onChange} />
            break
    }

    return (
        <div className={`${props.width ? 'inline' : 'block'}`}>
            { props.label &&
                <TextSmall muted className="uppercase mb-2">
                    {props.label ? props.label : props.name} { props.required && <span className="text-red-500">*</span>}
                </TextSmall>
            }
            {inputType}
        </div>
    )
}