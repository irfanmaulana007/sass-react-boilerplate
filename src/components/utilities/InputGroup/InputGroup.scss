@import './../../../Styles.scss';

input, select {
    border-bottom: 2px solid $theme-field;
    transition: .2s ease;
}

input:focus, select:focus {
    outline: 0!important;
    border-bottom: 2px solid $theme-blue;
}