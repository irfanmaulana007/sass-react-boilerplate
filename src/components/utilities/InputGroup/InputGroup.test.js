import React from 'react'
import { shallow } from 'enzyme'

import InputGroup from './InputGroup'

const props = {
    label: 'Input Email',
    name: 'email',
}
const wrapped = shallow(<InputGroup label={props.label} name={props.name} />)

describe('InputGroup', () => {
    it('should render the InputGroup Component correctly', () => {
        expect(wrapped).toMatchSnapshot()
    })
    it('renders the InputGroup Label', () => {
        expect(wrapped.find('label').text()).toEqual(props.label)
    })
})
