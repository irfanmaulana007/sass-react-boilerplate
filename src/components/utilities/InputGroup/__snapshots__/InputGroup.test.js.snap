// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup should render the InputGroup Component correctly 1`] = `ShallowWrapper {}`;
